app/controllers/provider/admin/account/logos_controller.rb

```diff
@@ -1,38 +1,33 @@
 class Provider::Admin::Account::LogosController < Provider::Admin::Account::BaseController
   activate_menu! :audience, :cms, :logo
+  before_action :authorize, only: %i[destroy update]
 
   def edit
     @profile = profile
   end
 
   def update
-    authorize! :update, :logo
-
-    redirect_to edit_provider_admin_account_logo_path and return false if params[:profile].blank?
-
-    profile.update_attribute(:logo, params[:profile][:logo])
-    @notice = 'Logo was successfully created.'
-
-    respond_to do |format|
-      format.html do
-        flash[:notice] = @notice
-        redirect_to edit_provider_admin_account_logo_path
-      end
-
-      format.js
-    end
+    update_logo(logo: params.require(:profile).require(:logo))
   end
 
   def destroy
-    authorize! :delete, :logo
+    update_logo(logo: nil)
+  end
 
-    profile.update_attribute(:logo, nil)
+  private
 
-    flash[:notice] = 'Your logo was successfully deleted.'
+  def update_logo(logo:)
+    if profile.update({logo: logo})
```

Comment:
why this change? `update` instead of `update_attribute` ?

Reply:
@macejmic because we have added validations in the model and they are skipped with `update_attribute` which is not what we want 😄 We want it to check the validations 😉

Reply:
i see. ok, makes sense. just didn't know what exactly does `update`.. needed to look to documentation 👯